mysql-test/suite/versioning/t/alter.test

```diff
@@ -469,5 +469,12 @@ alter table t add check (a > 0);
 insert into t values (0);
 insert into t values (2);
 
+--echo #
+--echo # MDEV-18869 Assertion `!((field)->vcol_info && (field)->stored_in_db())' failed in innodb_col_no upon altering table with system versioning
+--echo #
+create or replace table t1 (a int, b int generated always as (0) stored) engine=innodb with system versioning;
+set system_versioning_alter_history= keep;
+alter table t1 modify a int with system versioning;
```

Comment:
Please test also with a `generated…virtual` column, and with `ALGORITHM=COPY`.

I would prefer to have some data inside the table, and some `SELECT` to demonstrate that the operation not only does not crash, but actually is producing the correct results.

If you have some data inside the table, then
```sql
--enable_info
ALTER TABLE …;
--disable_info
```
will show "0 rows affected" if native ALTER was used, and a nonzero count if `ALGORITHM=COPY` was used.

Reply:
Fixed.